### PR TITLE
chore(tests): expose test utils (feature flag)

### PR DESCRIPTION
**Pull request by pront**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -857,6 +857,7 @@ disable-resolv-conf = []
 shutdown-tests = ["api", "sinks-blackhole", "sinks-console", "sinks-prometheus", "sources", "transforms-lua", "transforms-remap", "unix"]
 cli-tests = ["sinks-blackhole", "sinks-socket", "sources-demo_logs", "sources-file"]
 component-test-utils = []
+source-sender-test-utils = []
```
[the remark this answers is not in the document]
Comment (pront) [reply Cargo.toml:860]: Yes, this is a good idea. 

@sebastiantia did you want to expose more test utils?

[the remark this answers is not in the document]
Comment (sebastiantia) [reply Cargo.toml:860]: I have actually just opened up a PR to do just that https://github.com/vectordotdev/vector/pull/19864. I can close my PR or include your changes up to you...

[the remark this answers is not in the document]
Comment (pront) [reply Cargo.toml]: You can rebase on https://github.com/vectordotdev/vector/pull/19863/commits/37bfd6616f61a91d3d159fa0e81cd317346b50d3

**Review by bruceg — APPROVED** [review]

**Review by sebastiantia — APPROVED** [review]

**Review by neuronull — APPROVED** [review]

**Comment by neuronull** [comment]
note- the PR title doesn't match the semantic format (should probably be "chore(tests): <...>" . 

We currently don't have CI checks enforcing that. But it breaks the release generation script, requiring manual edit of the commit message for the release facilitator.

**Pull request merged**